### ISSUE TITLE
Use file access mode 'r' instead of 'r+' for readStream

### ISSUE THREAD

**Issue by pschirch** [thephpleague/flysystem#72]
Hi,

the [`Local Adapter`](https://github.com/thephpleague/flysystem/blob/master/src/Adapter/Local.php) uses file access mode `r+` in his [`readStream()`](https://github.com/thephpleague/flysystem/blob/master/src/Adapter/Local.php#L157) method.

The naming suggesting that a source is used read only. But `r+` allows [reading and writing](http://www.php.net/manual/en/function.fopen.php). In my current project I have only read access on the files, so following exception throws:

```
ContextErrorException: Warning: fopen(...): failed to open stream: Permission denied in .../vendor/league/flysystem/src/Adapter/Local.php line 157
```

The more suitable file access mode is `r`.

What's your opinion?

**Comment by barryvdh** [comment]
Is `rb` not more suitable? Shouldn't matter on Linux, but does in Windows, according to http://php.net/manual/en/function.fread.php#example-2648
> Warning: On systems which differentiate between binary and text files (i.e. Windows) the file must be opened with 'b' included in fopen() mode parameter.

(Via https://github.com/laravel/framework/pull/14319 by @BePsvPT )

**Comment by Frondor** [comment]
What's the actual status of this?

I'm getting this same error when trying to copy a folder 
`$manager->copy('local://'.$path, 'zip://'.$path);`

And I'm developing on windows 7, latest flysystem version.

**Comment by mirfilip** [comment]
The mode is `rb` for some time now - available since `1.0.25`, see #677. @frankdejonge maybe close this issue :)
